src/main/kotlin/com/github/avrokotlin/avro4k/schema/UnionSchemaFor.kt

```diff
@@ -17,7 +17,7 @@ class UnionSchemaFor(
 ) : SchemaFor {
     override fun schema(): Schema {        
         val leafSerialDescriptors =
-            descriptor.possibleSerializationSubclasses(serializersModule).sortedBy { it.serialName }
+            descriptor.possibleSerializationSubclasses(serializersModule).sortedBy { it.serialName }.toList()
```

Comment:
I would move the toList() in the next call (end of line 23) in order to save one iteration over all sequence elements.

Reply:
good catch